src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -863,6 +875,68 @@ public void testMethodTypingContext() throws Exception {
 	}
 	
 	@Test
+	public void testMethodTypingContextAdaptMethod() throws Exception {
+		Factory factory = build(new File("src/test/java/spoon/test/generics/testclasses"));
+		CtClass<?> ctClassLunch = factory.Class().get(Lunch.class);
+		CtMethod<?> trLunch_eatMe = ctClassLunch.filterChildren(new NameFilter<>("eatMe")).first();
+		CtClass<?> ctClassWeddingLunch = factory.Class().get(WeddingLunch.class);
+		CtMethod<?> trWeddingLunch_eatMe = ctClassWeddingLunch.filterChildren(new NameFilter<>("eatMe")).first();
+
+		MethodTypingContext methodSTH = new MethodTypingContext().setClassTypingContext(new ClassTypingContext(ctClassWeddingLunch));
```

Comment:
what does this line mean? I thought a MethodTypingContext would be always with respect to a specific method